### ISSUE TITLE
Behaviour of P0 is confusing when sounds are used.

### ISSUE THREAD

**Issue by jaustin** [microsoft/pxt-microbit#3740]
**Describe the bug**
For a user who wants to make sounds on the speaker there is not a good way to also use the pin touch events.

The following program explains the issue.

<img width="524" alt="image" src="https://user-images.githubusercontent.com/147113/99832223-2141aa00-2b58-11eb-8cc8-99f80652547f.png">

**To Reproduce**
* Have an 'on start' block that makes a sound
* Use an "on P0 pressed" handler

The P0 event will not fire after the sound.

Does not work (what the blocks generate)
```Typescript

input.onPinPressed(TouchPin.P0, function () {
    basic.showIcon(IconNames.No)
})
soundExpression.hello.play()
```
Works (manually modified)
```Typescript
soundExpression.hello.play()
input.onPinPressed(TouchPin.P0, function () {
    basic.showIcon(IconNames.No)
})
```

This would be fine if we had a way to un-direct sound to P0, but cheating and just mapping to a different pin doesn't work either (based on the above, this should work even if we didn't re-map the pin but the P0 event never fires here)
```Typescript
pins.setAudioPin(AnalogPin.P8)
soundExpression.hello.play()
input.onPinPressed(TouchPin.P0, function () {
    basic.showIcon(IconNames.No)
})


```

**Expected behavior**
If the user specifies touch on the pins on a V2 I think we should NOT map audio to them, so consider the pin 'claimed' when we bring up audio.

Alternatively, we could have an explicit way to specify you do not want audio on P0, so a "None" option in the set audio pin. This would need to actually decouple P0 gracefully though, which it appears it doesn't right now.

**Screenshots**
Add screenshots to help explain your problem. You can copy paste the screenshot in the github report. The .gif screen recording is very useful as well. 

**micro:bit version (please complete the following information):**

V2

**Comment by jaustin** [comment]
This has a fix in CODAL 0.2.21. If a sound handler is registered at any point, P0 is removed from the audio list. Because audio is lazily initialised, this means that sounds never end up going out on P0.
https://github.com/lancaster-university/codal-microbit-v2/commit/4989ce1bd66d0d3faa9064588b8de44714b87b57
Thanks @finneyj !

**Comment by martinwork** [comment]
https://makecode.microbit.org/_M20J3YJpXXrt

This is the original example, I think, with A also playing a giggle, and the set audio pin changed from P8 to P2

On live...

Hello is heard on P0, but the giggle isn't. Pressing B makes a click on P2, but the giggle isn't heard on P2. Touch on P0 doesn't work.

On beta...

No sound is heard on P0 and touch works on P0. Pressing B makes a click on P2, but the giggle isn't heard on P2.

Putting setAudioPin(AnalogPin.P2) at the top of the start block does not enable sounds on P2.

After also deleting the P0 handler, P2 plays hello, but not giggle.

After also deleting the P0 reference from the button A handler, P2 plays both hello and giggle.

The setAudioPin call makes the simulator show a sad face :(

**Comment by 4tronix** [comment]
Have just been debugging a customer issue, which turns out to be exactly this. Playing music (Microbit v1 or v2) stops events on Pin 0. You can re-enable pin edge events with a block after each music piece, but not very pleasant.
The Set Audio Pin is currently broken and gives error "pxsim.pins.setAudioPin is not a function". Setting audio to something other than Pin 0 should free up Pin 0 to work correctly, but because of this second bug it doesn't.

**Comment by microbit-mark** [comment]
> micro:bit support #41097

Also raised on support by someone with accessories that make use of P0 on the edge connector.

**Comment by finneyj** [comment]
Hi folks. This patch is already implemented here:
https://github.com/lancaster-university/codal-microbit-v2/commit/7145889d5d66325cce67cb2cb26f480dc46f8e5a

It's also included in the v0.2.21 release, which I think went to MakeCode beta last week.

**Comment by pelikhan** [comment]
Fix for setAudioPin sim is in ``v3.0.63``

**Comment by microbit-mark** [comment]
> micro:bit support 42348

Another person has been in touch to request having an option to disable sound output on Pin0 completely, rather than retargeting it to another pin.

**Comment by 4tronix** [comment]
I fully support this request. We have a number of products that use all 
the available pins and it would be good to be able to also use the sound 
on the Microbit, so disabling music output to the pins is necessary for 
this.

Gareth

------ Original Message ------

**Comment by microbit-mark** [comment]
It would be good to have an `edge connector sound on/off` block in a similar way to speaker on/off. We could also update the simulator when this block is used to hide/strike-through the headphone connection image

**Comment by abchatra** [comment]
Fixed by setAudioPin API? If not let me know, I will reopen this ticket.

**Comment by microbit-mark** [comment]
@abchatra I think the issue is fixed, so I've opened #3937 to consider having a way to disable edge connector sound
